header.go

```diff
@@ -2247,7 +2247,11 @@ var (
 	errSmallBuffer = errors.New("small read buffer. Increase ReadBufferSize")
 )
 
-type errNothingRead struct {
+// ErrNothingRead is returned that means just a keep-alive connection
+// closing down either because the remote closed it or because
+// or a read timeout on our side.
+// Either way just close the connection and don't return any error response.
```

Comment:
```suggestion
// ErrNothingRead is returned when a keep-alive connection is closed,
// either because the remote closed it or because of a read timeout.
```